data/summary/types/cgm_test.go

```diff
@@ -86,8 +81,8 @@ func NewDataSetCGMDataRanges(startTime time.Time, hours float64, ranges DataRang
 		for i, bracket := range glucoseBrackets {
 			datumTime := startTime.Add(time.Duration(-(count+i+1))*time.Minute*5 - gapCompensation)
 
-			datum := NewGlucose(typ, pointer.FromString(units), &datumTime, &deviceId, &uploadId)
-			datum.Value = pointer.FromFloat64(bracket[0] + (bracket[1]-bracket[0])*rand.Float64())
+			datum := NewGlucose(typ, unit, &datumTime, &deviceId, &uploadId)
+			datum.Value = pointer.FromAny(bracket[0] + (bracket[1]-bracket[0])*rand.Float64())
```

Comment:
This is pretty clearly a float64, no? Why not use pointer.FromFloat64? I like types, they help keep my code safe and alert me to problems early on.

Reply:
Types are still validated, this is a generic function and does not return a typeless interface, closing as discussed.